vscode-extension/src/extension.ts

```diff
@@ -26,6 +32,46 @@ export function activate(context: vscode.ExtensionContext) {
   });
   context.subscriptions.push(setupCommand);
 
+  let shareCommand = vscode.commands.registerCommand(COMMANDS.SHARE, async () => {
+    // TODO: use vscode.window.activeTextEditor instead of hardcoding
+    const fileName: string = "/Users/rossdancraig/Projects/aiconfig/cookbooks/Gemini/gemini_demo.aiconfig.json";
+    const sanitizedFileName: string = sanitizeFileName(fileName);
+
+    // Use empty accessKeyId and secretAccessKey so bucket can be accessed publicly
+    AWS.config.update({
+      accessKeyId: '',
+      secretAccessKey: '',
+    });
+    const s3 = new AWS.S3();
+
+    const randomPath = Math.round(Math.random() * 10000);
+    const bucket: string = "lastmileai.aiconfig.public";
+    const uploadKey: string = `aiconfigs/${getTodayDateString()}/${randomPath}/${sanitizedFileName}`;
+    const configString = readFileSync(fileName, 'utf-8');
+    // TODO: Will also need to check for yaml files and change the contentType accordingly
+    const contentType = "application/json";
+
+    const input = {
+      ACL: 'public-read',
+      ContentType: contentType,
+      // TODO: Check if we can just pass in fileName itself? Don't think so
+      Body: configString, 
+      Bucket: bucket,
+      Key: uploadKey,
+    };
+    await s3.putObject(input).promise();
+    
+    const s3Url: string = `https://s3.amazonaws.com/${bucket}/${uploadKey.replace(/[ ]/g, "%20")}`
+    
+    const lastmileUploadUrl: string = LASTMILE_BASE_URI + "api/aiconfig/upload";
+    const response = await ufetch.post(lastmileUploadUrl, {"url": s3Url, "source": "vscode"});
+    if (response?.id !== null && response?.id !== undefined) {
+      const renderingUrl: string = LASTMILE_BASE_URI + `aiconfig/${response?.id}`;
+      vscode.window.showInformationMessage("shareUri: " + renderingUrl);
```

Comment:
Will the share button also show its own modal on top of this?

Reply:
Oh, I guess this is unrelated to the button at all, just a command palette command. Nvm